src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileDirectoryStep.java

```diff
@@ -74,23 +72,18 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
                     .datasetId(datasetId)
                     .loadTag(loadModel.getLoadTag());
                 fileDao.createDirectoryEntry(dataset, newEntry);
-            } else {
-                if (!StringUtils.equals(existingEntry.getLoadTag(), loadModel.getLoadTag())) {
-                    // (a) Exists and is not our file
-                    throw new FileAlreadyExistsException("Path already exists: " + targetPath);
-                }
-
-                // (b) or (c) Load tags match - check file ids
+            } else if (StringUtils.equals(existingEntry.getLoadTag(), loadModel.getLoadTag())) {
+                // 2. (a) or (b) Load tags match - check file ids
                 if (!StringUtils.equals(existingEntry.getFileId(), fileId)) {
-                    // (c) We are in a re-run of a load job. Try to get the file entry.
+                    // (b) We are in a re-run of a load job. Try to get the file entry.
                     fileId = existingEntry.getFileId();
                     workingMap.put(FileMapKeys.FILE_ID, fileId);
                     FireStoreFile fileEntry = fileDao.lookupFile(dataset, fileId);
                     if (fileEntry != null) {
-                        // (c)(i) We successfully loaded this file already
+                        // (b)(i) We successfully loaded this file already
                         workingMap.put(FileMapKeys.LOAD_COMPLETED, true);
                     }
-                    // (c)(ii) We are recovering and should continue this load; leave load completed false/unset
+                    // (b)(ii) We are recovering and should continue this load; leave load completed false/unset
                 }
 
```

Comment:
> I think we should still include an "else" option that throws a step failure so that this doesn't fail silently…. But, then we're right back to where we started.

I might be misunderstanding how steps work, but how would it get into this state?

Reply:
Adding an error on `else` would change the current behavior, so it may be out of scope for this change. The previous code "handled" that case by leaving `LOAD_COMPLETE` false. The theory is that this means the flight is in recovery so the case isn't an error.

Reply:
Maybe it would be worth adding a comment about what the next steps are if the flight is in recovery? but I agree that it's out of the scope.

I find this workflow really confusing, but appreciate the documentation

Reply:
Ohh, I see. I thought the previous code handled the "else" condition, but I was reading it incorrectly.